integration-test/background/click-to-load-facebook.js

```diff
@@ -26,61 +70,74 @@ describe('Test Facebook Click To Load', () => {
 
-    it('CTL: Should block Facebook requests by default', async () => {
+    it('CTL: Facebook request blocking/redirecting', async () => {
+        // Open the test page and start logging network requests.
         const page = await browser.newPage()
-
-        try {
-            await page.goto(testSite, { waitUntil: 'networkidle2', timeout: 10000 })
-        } catch (e) {
-            // timed out waiting for page to load, let's try running the test anyway
-            console.log(`Timed out: ${e}`)
+        const pageRequests = []
+        logPageRequests(page, pageRequests)
+
+        // Initially there should be a bunch of requests. The SDK should
+        // be redirected to our surrogate but otherwise Facebook requests should
+        // be blocked.
+        await page.goto(testSite, { waitUntil: 'networkidle0' })
+        await page.waitForNetworkIdle({ idleTime: 1000 })
+        {
+            const {
+                requestCount, blockCount, allowCount, facebookSDKRedirect
+            } = summariseFacebookRequests(pageRequests)
+
+            expect(facebookSDKRedirect.checked).toBeTrue()
+            expect(facebookSDKRedirect.alwaysRedirected).toBeTrue()
+            expect(requestCount).toBeGreaterThan(5)
+            expect(blockCount).toEqual(requestCount)
+            expect(allowCount).toEqual(0)
         }
 
-        // give it little time just to be sure (facebook widgets can take time to load)
-        await page.waitForTimeout(4000)
-        const fbRequestData = await page.evaluate(() => {
-            return {
-                requests: document.getElementById('facebook_call_count').innerHTML,
-                expectedCalls: document.getElementById('facebook_iFrames').innerHTML
-            }
+        // Once the user clicks to load the Facebook content, the SDK should be
+        // loaded and the content should be unblocked.
+        pageRequests.length = 0
+        const buttonCount = await page.evaluate(() => {
+            window.buttons =
+                Array.from(document.querySelectorAll('body > div'))
+                    .map(div => div.shadowRoot && div.shadowRoot.querySelector('button'))
+                    .filter(button => button)
+            return window.buttons.length
         })
-
-        expect(Number(fbRequestData.requests)).toBeLessThanOrEqual(Number(fbRequestData.expectedCalls))
-
-        try {
-            await page.close()
-        } catch (e) {}
-    })
-
-    xit('CTL: Should load Facebook elements on click', async () => {
-        const page = await browser.newPage()
-
-        try {
-            await page.goto(testSite, { waitUntil: 'networkidle2', timeout: 10000 })
-        } catch (e) {
-            // timed out waiting for page to load, let's try running the test anyway
-            console.log(`Timed out: ${e}`)
+        for (let i = 0; i < buttonCount; i++) {
+            const button = await page.evaluateHandle(i => window.buttons[i], i)
+            await button.click()
```

Comment:
do we care if those are clicked one by one? WHy not click all at once?

Reply:
Well, no I don't think we care if they are clicked one-by-one, or what order they're clicked. In practice the buttons are clicked very fast however and I didn't see the point in optimising it. The main time taken by these tests is waiting for requests. Happy to rework to use `Promise.all` if you like?

Reply:
If there is no time difference, I don't mind keeping current solution 👍